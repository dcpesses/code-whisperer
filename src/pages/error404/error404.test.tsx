import {render} from '@testing-library/react';
import Error404 from './index';

interface LinkProps {
  className: string,
  to: string,
  children: React.ReactNode
}

vi.mock('react-router-dom', () => {
  const reactRouterDom = vi.importActual('react-router-dom');
  return {
    ...reactRouterDom,
    Link: ({className, to, children}: LinkProps) => (
      <div className={className} data-to={to}>
        {children}
      </div>
    )
  };
});

describe('Error404', () => {
  test('Should render as expected', () => {
    const {container} = render(<Error404 />);
    expect(container).toMatchSnapshot();
  });
});
