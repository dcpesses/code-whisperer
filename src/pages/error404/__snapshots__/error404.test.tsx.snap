// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error404 > Should render as expected 1`] = `
<div>
  <div
    class="error-404 full-pg d-flex w-100 h-100 my-0 mx-auto flex-column"
  >
    <main
      class="px-3 text-center"
    >
      <h1
        class="display-1 ppb-4"
      >
        404
      </h1>
      <p>
        <small
          class="fst-italic"
        >
          I'm sorry, Dave. I'm afraid I can't do that.
        </small>
      </p>
      <div
        class="col-6 mx-auto"
      >
        <div
          class="btn btn-outline-primary btn-lg mt-5"
          data-to="/"
        >
          Go Home
        </div>
      </div>
    </main>
  </div>
</div>
`;
